HACKING.rst

```diff
@@ -0,0 +1,80 @@
+*********
```

Comment:
I see this gets deleted in a subsequent commit.  Do you think it could be adapted for craft-cli?  I think it contains a lot of useful info about the tooling that isn't in the original README, which is minimal and outdated (like using `Make`)

Reply:
You're right! I thought about this a bit and I think that a good 'compromise' might be to finish this merge and then gave a separate issue/PR to update craft-cli's README and HACKING to integrate the content from starbase, because as you point out it's quite useful. What do you think?

Reply:
Yes, a separate issue/PR is fine with me!

Reply:
Done, #152